statsmodels/formula/_manager.py

```diff
@@ -0,0 +1,723 @@
+            elif isinstance(data, np.rec.recarray):
+                # workaround no recarray support in formulaic
+                _data = pd.DataFrame.from_records(data)
+            else:
+                _data = data
+
+            _eval_env = eval_env
+            if HAVE_PATSY:
+                from patsy.eval import EvalEnvironment
+
+                if isinstance(eval_env, EvalEnvironment):
+                    warnings.warn(EVAL_ENV_WARNING, FutureWarning)
+
+                    ns = eval_env._namespaces
+                    _eval_env = {}
+                    for val in ns:
+                        _eval_env.update(val)
+            if not isinstance(_eval_env, (int, dict)):
+                raise TypeError("context (eval_env) must be an int or a dict.")
+
+            _ordering = statsmodels.formula.options.ordering
+            if isinstance(formula, self.model_spec_type):
+                _formula = formula
+            if _ordering == "legacy":
+                _formula = self._legacy_orderer(formula, _data, context=_eval_env)
+            else:
+                feature_flags = formulaic.parser.DefaultParserFeatureFlag.TWOSIDED
+                parser = formulaic.parser.DefaultFormulaParser(
+                    feature_flags=feature_flags
+                )
+                _formula = formulaic.formula.Formula(
+                    formula, _ordering=_ordering, _parser=parser
+                )
+            if prediction:
+                if hasattr(_formula, "rhs"):
+                    _formula = _formula.rhs
+            original_index = _data.index
+            na_drop = "na_action" in kwargs and kwargs["na_action"] == "drop"
+            if na_drop:
+                _data = _data.copy(deep=False)
+                _data.index = pd.RangeIndex(_data.shape[0])
+
+            output = formulaic.model_matrix(
+                _formula, _data, context=_eval_env, **kwargs
+            )
+            if na_drop:
+                if isinstance(output, ModelMatrices):
+                    rhs = output.rhs
+                else:
+                    rhs = output
+                if rhs.shape[0] == _data.shape[0]:
+                    replacement_index = original_index
+                    self._missing_mask = pd.Series(
+                        np.full(rhs.shape[0], False), index=original_index, name=None
+                    )
+                else:
+                    replacement_index = original_index[rhs.index]
+                    self._missing_mask = (
+                        pd.Series(rhs.index, index=rhs.index)
+                        .reindex(pd.RangeIndex(_data.shape[0]))
+                        .isna()
+                    )
+                for key in ("lhs", "rhs", "root"):
+                    if hasattr(output, key):
+                        df = getattr(output, key)
+                        df.index = replacement_index
+            if isinstance(output, formulaic.ModelMatrices):
+                if (
```

Comment:
You can now just do: `if not isinstance(SimpleFormula):`

Reply:
Do you mean `isinstance(output.model_spec.formula, SimpleFormula)`?